pipeline/ffmpeg.go

```diff
@@ -115,7 +116,20 @@ func (f *ffmpeg) HandleStartUploadJob(job *JobInfo) (*HandlerOutput, error) {
 		return nil, fmt.Errorf("error downloading source manifest: %s", err)
 	}
 
-	job.sourceSegments = len(sourceManifest.GetAllSegments())
+	sourceSegments := sourceManifest.GetAllSegments()
+	job.sourceSegments = len(sourceSegments)
+
+	if job.sourceSegments > 0 {
+		firstSeg := sourceSegments[0]
+		lastSeg := sourceSegments[job.sourceSegments-1]
+
+		if err := probeSourceSegment(firstSeg, transcodeRequest.SourceManifestURL); err != nil {
+			return nil, err
+		}
+		if err := probeSourceSegment(lastSeg, transcodeRequest.SourceManifestURL); err != nil {
```

Comment:
We could be more efficient and check it's not a single segment video before doing this, but let's 🚢 🇮🇹  for now